pkg/storegateway/bucket.go

```diff
@@ -899,13 +902,16 @@ func (s *BucketStore) Series(req *storepb.SeriesRequest, srv storepb.Store_Serie
 			if s.maxSeriesPerBatch <= 0 {
 				// When streaming is disabled, the time spent iterating over the series set is the
 				// actual time spent merging series and sending response to the client.
-				stats.synchronousSeriesMergeDuration += time.Since(begin)
+				stats.synchronousSeriesMergeDuration += time.Since(iterationBegin)
 			} else {
 				// When streaming is enabled, the time spent iterating over the series set is the
-				// actual time spent fetching series and chunks, and sending them to the client.
-				// We split the two timings to have a better view over how time is spent.
+				// actual time spent fetching series and chunks, encoding and sending them to the client.
+				// We split the timings to have a better view over how time is spent.
 				stats.streamingSeriesFetchSeriesAndChunksDuration += stats.streamingSeriesWaitBatchLoadedDuration
-				stats.streamingSeriesSendResponseDuration += time.Duration(util_math.Max64(0, int64(time.Since(begin)-stats.streamingSeriesFetchSeriesAndChunksDuration)))
+				stats.streamingSeriesEncodeResponseDuration += encodeDuration
+				stats.streamingSeriesSendResponseDuration += sendDuration
+				stats.streamingSeriesOtherDuration += time.Duration(util_math.Max64(0, int64(time.Since(iterationBegin)-
```

Comment:
Note to reviewers: "other" is to track all other timing to iterate over series, but not actually spent in any specific operation. It's just to make sure we're not missing anything.